src/LinuxTracing/SwitchesStatesNamesVisitorTest.cpp

```diff
@@ -389,34 +396,57 @@ TEST_F(SwitchesStatesNamesVisitorTest,
 }
 
 void SwitchesStatesNamesVisitorTest::ProcessFakeEventsForThreadStateTests() {
-  // kTid1
+  // kPid1, kTid1
   visitor_.ProcessInitialState(kStartTimestampNs, kTid1, 'D');
+  visitor_.ProcessInitialState(kStartTimestampNs, kTid3, 'D');
+  (void)kComm2;
+  (void)kCpu1;
+  (void)kCpu2;
   PerfEvent{MakeFakeSchedWakeupPerfEvent(kTid1, kWakeTimestampNs1)}.Accept(&visitor_);
   PerfEvent{MakeFakeSchedSwitchPerfEvent(kCpu1, kPrevTid, kPrevTid, kRunnableStateMask, kTid1,
                                          kInTimestampNs1)}
       .Accept(&visitor_);
   PerfEvent{
-      MakeFakeSchedSwitchPerfEvent(kCpu1, kPid, kTid1, kDeadStateMask, kNextTid, kOutTimestampNs1)}
+      MakeFakeSchedSwitchPerfEvent(kCpu1, kPid1, kTid1, kDeadStateMask, kNextTid, kOutTimestampNs1)}
       .Accept(&visitor_);
 
-  // kTid2
+  // kPid1, kTid2
   PerfEvent{MakeFakeTaskNewtaskPerfEvent(kTid2, kComm2, kNewTimestampNs2)}.Accept(&visitor_);
   PerfEvent{MakeFakeSchedSwitchPerfEvent(kCpu2, kPrevTid, kPrevTid, kRunnableStateMask, kTid2,
                                          kInTimestampNs2)}
       .Accept(&visitor_);
-  PerfEvent{MakeFakeSchedSwitchPerfEvent(kCpu2, kPid, kTid2, kInterruptibleSleepStateMask, kNextTid,
-                                         kOutTimestampNs2)}
+  PerfEvent{MakeFakeSchedSwitchPerfEvent(kCpu2, kPid1, kTid2, kInterruptibleSleepStateMask,
+                                         kNextTid, kOutTimestampNs2)}
       .Accept(&visitor_);
   PerfEvent{MakeFakeSchedWakeupPerfEvent(kTid2, kWakeTimestampNs2)}.Accept(&visitor_);
 
+  // kPid2, kTid3
+  PerfEvent{MakeFakeSchedWakeupPerfEvent(kTid3, kWakeTimestampNs1)}.Accept(&visitor_);
+  PerfEvent{MakeFakeSchedSwitchPerfEvent(kCpu3, kPrevTid, kPrevTid, kRunnableStateMask, kTid3,
+                                         kInTimestampNs1)}
+      .Accept(&visitor_);
+  PerfEvent{
+      MakeFakeSchedSwitchPerfEvent(kCpu3, kPid2, kTid3, kDeadStateMask, kNextTid, kOutTimestampNs1)}
+      .Accept(&visitor_);
+
+  // kPid2, kTid4
+  PerfEvent{MakeFakeTaskNewtaskPerfEvent(kTid4, kComm3, kNewTimestampNs2)}.Accept(&visitor_);
+  PerfEvent{MakeFakeSchedSwitchPerfEvent(kCpu4, kPrevTid, kPrevTid, kRunnableStateMask, kTid4,
+                                         kInTimestampNs2)}
+      .Accept(&visitor_);
+  PerfEvent{MakeFakeSchedSwitchPerfEvent(kCpu4, kPid2, kTid4, kInterruptibleSleepStateMask,
+                                         kNextTid, kOutTimestampNs2)}
+      .Accept(&visitor_);
+  PerfEvent{MakeFakeSchedWakeupPerfEvent(kTid3, kWakeTimestampNs2)}.Accept(&visitor_);
+
```

Comment:
You are reusing some of the timestamp constants. So events are processed out of order here, which is not the intent of this test. If you want to keep this part, new timestamps need to be defined.

Reply:
> Should we still define new timestamps?

I think so. This and the other `PerfEventVisitor`s assume that events come in order, which is guaranteed by `PerfEventProcessor`/`PerfEventQueue`. We should honor that in the tests, even if in some cases it works anyway.
It also makes the test more readable, because the idea of the names here is that a `k...TimestampNs#` corresponds to thread `kTid#`.
To simplify a bit and also keep the number of constants under control, you could consider keeping one process with two threads and one process with only one thread.

Reply:
ok, done and done